### DISCUSSION TITLE
Same cache tag for different queries (revalidateTag)

### DISCUSSION THREAD

**Discussion by martin-coded** [vercel/next.js#56205] [Help] [answered]
### Summary

I am a little lost with the revalidateTag topic.

Is it possible to tag different queries with the same tag, so all the queries with the same tag are invalidated? This would be really helpful, but I am not sure if this is intended to do so.

myTag1 -> query /a -> used in pageB, pageC
myTag1 -> query /b -> used in pageA, pageB 
myTag2 -> query /c -> used in pageA, PageC

revalidateTag(myTag1) invalidates query /a and /b so pageA, pageB and pageC will be updated updated?
revalidateTag(myTag2) invalidates query /c so pageA and pageC will be updated?

Best Regards,
Martin

**Comment by leerob** [comment, marked as answer]
If you want to invalidate multiple fetches, then all fetches must have been tagged with the same cache tag.

```js
const res1 = await fetch('https://.../one', { next: { tags: ['collection'] } })
const res2 = await fetch('https://.../two', { next: { tags: ['collection'] } })
const res3 = await fetch('https://.../three', { next: { tags: ['collection'] } })

revalidateTag('collection') // revalidates all three fetches
```

https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating#on-demand-revalidation
